Modules/Quotas/test/Quotas.Application.Tests/Tests/Quotas/CreateQuotaForIdentity/HandlerTests.cs

```diff
@@ -49,6 +52,37 @@ public async Task Creates_quota_for_identity()
         ).MustHaveHappened();
     }
 
+    [Fact]
+    public async Task Create_quota_with_invalid_metric_key_throws_domain_exception()
+    {
+        // Arrange
+        var command = new CreateQuotaForIdentityCommand(IdentityAddress.Parse("id1KJnD8ipfckRQ1ivAhNVLtypmcVM5vPX4j"), "An-Invalid-Metric-Key", 5, QuotaPeriod.Month);
+        var identitiesRepository = A.Fake<IIdentitiesRepository>();
+        var metricsRepository = new FindMetricsStubRepository(new Metric(MetricKey.NumberOfSentMessages, "Number Of Sent Messages"));
+        var handler = CreateHandler(identitiesRepository, metricsRepository);
+
+        // Act
+        var exception = await Assert.ThrowsAsync<DomainException>(async () => await handler.Handle(command, CancellationToken.None));
+        exception.Message.Should().Contain("The given metric key is not supported.");
```

Comment:
Since the error code is expressive enough, there is no need to check for the error message